Data/NF/Unsafe.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 708
+{-# LANGUAGE PatternSynonyms #-}
+#endif
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Data.NF.Unsafe
+-- Copyright   :  (c) David Feuer 2016
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  ezyang@cs.stanford.edu
+-- Stability   :  experimental
+-- Portability :  portable
+--
+-- This module provides an unsafe pattern synonym for the type 'NF', allowing you
+-- to create values of type 'NF' without any runtime overhead if you fulfill a
+-- proof obligation that the value is already in normal form. It also provides
+-- functions allowing more general coercions relating to 'NF'.
```

Comment:
For what to be imported qualified? There should be no need for qualified
imports unless someone imports .Internal *and* .Unsafe, and I don't think
there's any conceivable reason to do that!

On Jul 22, 2016 6:10 PM, "Edward Z. Yang" <notifications@github.com> wrote:

> In Data/NF/Unsafe.hs
> <https://github.com/ezyang/nf/pull/6#discussion_r71953362>:
>
> > +#endif
> > +
> > +-----------------------------------------------------------------------------
> > +-- |
> > +-- Module      :  Data.NF.Unsafe
> > +-- Copyright   :  (c) David Feuer 2016
> > +-- License     :  BSD-style (see the file LICENSE)
> > +--
> > +-- Maintainer  :  ezyang@cs.stanford.edu
> > +-- Stability   :  experimental
> > +-- Portability :  portable
> > +--
> > +-- This module provides an unsafe pattern synonym for the type 'NF', allowing you
> > +-- to create values of type 'NF' without any runtime overhead if you fulfill a
> > +-- proof obligation that the value is already in normal form. It also provides
> > +-- functions allowing more general coercions relating to 'NF'.
>
> I guess the intent for this function is for it to be imported qualified?
> Needs a note about it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ezyang/nf/pull/6/files/db57fa061540b46fb54928e8131e84be376b7fe9#r71953362>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABzi_bmBWv3FRXniHzWXCe1HiJknxoM-ks5qYT_ogaJpZM4JTKLc>
> .
>